binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtilsTests.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.kafka.streams;
+
+import java.util.Date;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.core.ResolvableType;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Unit tests for {@link SerdeResolverUtils}.
+ *
+ * @author Chris Bono
+ */
+class SerdeResolverUtilsTests {
+
+	/**
+	 * Verify that {@link SerdeResolverUtils#findMatchingSerdes} returns the proper serdes
+	 * in the proper order for the following grid:
+	 * <p><br>
+	 * <pre>{@code
+	 * ------------------------------------------------------------------
+	 * KStream type       | Serde type
+	 * ------------------------------------------------------------------
+	 *                    | GE<Date> | GE<? extends Date> | GE<?> | GE
+	 * ------------------------------------------------------------------
+	 * GE<Date>           | Y        | Y                  | Y     | Y
+	 * GE<? extends Date> | N        | Y                  | Y     | N
+	 * GE<?>              | N        | N                  | Y     | N
+	 * GE                 | N        | N                  | N     | N
+	 * ------------------------------------------------------------------
+	 * }</pre>
+	 */
+	@Test
```

Comment:
In follow-on PR I will also beef up this test to include a more complicated Serde type that has 2 level generics as well as some edge-casey like checks.

Reply:
To be handled in #2370